### ISSUE TITLE
NoClassDefFoundError or build error with a third-party library

### ISSUE THREAD

**Issue by EugeneSytnyk** [newrelic/newrelic-android-agent#145]
When integrating the New Relic agent on Android, I encountered a compatibility issue with the library `org.bouncycastle:bcprov-jdk15to18:1.69`, which is used by a third-party SDK. There are two different problems for versions 7.0.0 and 7.1.0:
For version 7.0.0, the problem appears as follows: The build completes successfully, and the application is built. However, when executing code from the third-party SDK, we encounter an exception:
```
Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lorg/bouncycastle/jcajce/provider/asymmetric/rsa/PSSSignatureSpi$SHA256withRSA;...
Caused by: java.lang.ClassNotFoundException: Didn't find class "org.bouncycastle.jcajce.provider.asymmetric.rsa.PSSSignatureSpi$SHA256withRSA" on path: DexPathList...
```
Everything works fine without New Relic enabled.
For version 7.1.0, the problem appears as follows: An error occurs during the build:
```
Execution failed for task ':app:newrelicTransformClassesForDebug'.> SHA-256 digest error for org/bouncycastle/asn1/ASN1GeneralizedTime.class
*Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:newrelicTransformClassesForDebug'....

```
Android version:
minSdk: 23
targetSdk: 33
New Relic version:
7.0.0, 7.1.0

**Comment by ywang-nr** [comment]
@EugeneSytnyk Thanks for the detailed information, we are aware of this issue and have proposed fix. 

But we were unable to reproduce it with just the bouncycastle library, would you help to share:
1. What is the third-party SDK you are using in your app? 
2. Is it possible to share sample code snippet using the third party library?

Thank you.

**Comment by EugeneSytnyk** [comment]
1. It is `com.adyen.checkout:3ds2:4.12.1`
2. For the first case with version 7.0.0 it will not be possible to provide a code example, since this error occurs when making a payment. For the second case with version 7.1.0, I will try to make a sample code and attach it here

Thank you

**Comment by ywang-nr** [comment]
Fixed in [v7.2.0](https://github.com/newrelic/newrelic-android-agent/releases/tag/7.2.0)
